File: DavidRey_GenerarTiempos.py
import time
# se crea la funcion de fibonacci de forma recursiva
def fibonacci(N):
	#casos bases	
	if(N==0):
		return(0.0)
	elif(N==1):
		return(1.0)
	#Caso recursivo
	else:
		return(fibonacci(N-1)+fibonacci(N-2))
# se crea funcion del tiempo
def get_time(N):
	t0 = time.time()
	fibonacci(N)
	t1 = time.time()-t0
	return(t1)
# se imprimen los tiempos hasta fibonacci de 34
for i in range(35):
		print(get_time(i))
		

File: DavidRey_GenerarTiempos.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>


using std::cout;
using std::endl;
//inicializo las funciones
int fibonacci(int N);
float get_time(int N);
// Construyo la funcion de fibonnaci
int fibonacci(int N){

//se hacen los casos base
if(N==0){
	return(0);
}
else if(N==1){
	return(1);
}
//llamo recursivamente la funcion
else{
	return (fibonacci(N-1)+fibonacci(N-2));
}
}
//construyo la funcion del tiempo

float get_time(int N){
clock_t t;
t = clock();
fibonacci(N);
t = clock() - t;
float time = ((float)t)/CLOCKS_PER_SEC;
	return(time);
}

// se hace el main 
int main(){
//declari variable de tiempo que se va a devolver
float t=0;
//se suma al t el tiempo que toma calcular fibonacci para cada i hasta 34
for(int i = 0; i < 35; ++i){
t=t+get_time(i);
}
cout <<t<< endl;
return 0;
}

